File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/exam/exam_participant_response.dart
class ExamParticipantResponse {
  final int id;
  final int userId;
  final String fullName;
  final String className;
  final String status;
  final int score;

  ExamParticipantResponse({
    required this.id,
    required this.userId,
    required this.fullName,
    required this.className,
    required this.status,
    required this.score,
  });

  factory ExamParticipantResponse.fromJson(Map<String, dynamic> json) {
    return ExamParticipantResponse(
      id: json['id'],
      userId: json['userId'],
      fullName: json['fullName'] ?? '',
      className: json['className'] ?? '',
      status: json['status']?.toString() ?? '',
      score: json['score'] ?? 0,
    );
  }
}

File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/ExamParticipantResponse.java
package org.example.olympic_ot_project.dto.exam;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.example.olympic_ot_project.core.ParticipantStatus;


@Data
@AllArgsConstructor
public class ExamParticipantResponse {
    private Integer id;
    private Integer userId;
    private String fullName;
    private String className;
    private ParticipantStatus status;
    private Integer score;
    Integer seatNumber;
}
